### PR TITLE
[BEAM-1768] Fix assert_that for empty inputs

### PR DESCRIPTION
**Pull request by robertwb**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by robertwb** [comment]
R: @vikkyrk

File: sdks/python/apache_beam/transforms/util.py [hunk 1]
```diff
@@ -222,12 +223,17 @@ def assert_that(actual, matcher, label='assert_that'):
...
-              | "UnKey" >> Map(lambda (k, v): v)
-              | "Match" >> Map(matcher))
+      # We must have at least a single element to ensure the matcher
+      # code gets run even if the input pcollection is empty.
+      keyed_singleton = pcoll.pipeline | Create([(None, None)])
+      keyed_actual = (
+          pcoll
+          | WindowInto(window.GlobalWindows())
+          | "ToVoidKey" >> Map(lambda v: (None, v)))
+      _ = ((keyed_singleton, keyed_actual)
```
Comment (vikkyrk) [inline sdks/python/apache_beam/transforms/util.py]: Why the `_`?

Comment (robertwb) [inline sdks/python/apache_beam/transforms/util.py]: Makes lint happy with both the unusedness and warning.

**Comment by robertwb** [comment]
Thanks.

**Pull request closed without merging**